torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,410 @@
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask
+        """
+        # See https://github.com/VSainteuf/pastis-benchmark/blob/main/code/dataloader.py#L201 # noqa: E501
+        # even though the mask file is 3 bands, we just select the first band
+        array = np.load(self.files[index]["semantic"])[0].astype(np.uint8)
+        tensor = torch.from_numpy(array).long()
+        return tensor
+
+    def _load_instance_targets(self, index: int) -> tuple[Tensor, Tensor, Tensor]:
+        """Load the instance segmentation targets for a single sample.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the instance segmentation mask, box, and label for each instance
+        """
+        mask_array = np.load(self.files[index]["semantic"])[0].astype(np.uint8)
+        instance_array = np.load(self.files[index]["instance"])
+
+        mask_tensor = torch.from_numpy(mask_array)
+        instance_tensor = torch.from_numpy(instance_array)
+
+        # Convert instance mask of N instances to N binary instance masks
+        instance_ids = torch.unique(instance_tensor)
+        # Exclude a mask for unknown/background
+        instance_ids = instance_ids[instance_ids != 0]
+        instance_ids = instance_ids[:, None, None]
+        masks: Tensor = instance_tensor == instance_ids
+
+        # Parse labels for each instance
+        labels_list = []
+        for mask in masks:
+            label = mask_tensor[mask]
+            label = torch.unique(label)[0]
+            labels_list.append(label)
+
+        # Get bounding boxes for each instance
+        boxes_list = []
+        for mask in masks:
+            pos = torch.where(mask)
+            xmin = torch.min(pos[1])
+            xmax = torch.max(pos[1])
+            ymin = torch.min(pos[0])
+            ymax = torch.max(pos[0])
+            boxes_list.append([xmin, ymin, xmax, ymax])
+
+        masks = masks.to(torch.uint8)
+        boxes = torch.tensor(boxes_list).to(torch.float)
+        labels = torch.tensor(labels_list).to(torch.long)
+
+        return masks, boxes, labels
+
+    def _load_files(self) -> list[dict[str, str]]:
+        """List the image and target files.
+
+        Returns:
+            list of dicts containing image and semantic/instance target file paths
+        """
+        self.idxs = []
+        metadata_fn = os.path.join(self.root, self.directory, "metadata.geojson")
+        with fiona.open(metadata_fn) as f:
+            for row in f:
+                fold = int(row["properties"]["Fold"])
+                if fold in self.folds:
+                    self.idxs.append(row["properties"]["ID_PATCH"])
+
+        files = []
+        for i in self.idxs:
+            suffix = f"{i}.npy"
+            files.append(
+                dict(
+                    s2=os.path.join(self.root, self.directory, self.prefix["s2"])
+                    + suffix,
```

Comment:
Lots of duplication here. I would use something like:
```python
path = os.path.join(self.root, self.directory, "{}") + i + ".npy"
files.append({
    "s2": path.format(self.prefix["s2"]), 
    ...
})
```
That will allow each key/value pair to fit on a single line too.